### PR TITLE
Update HLRBRep_AlgoPy.xml

### PR DESCRIPTION
**Pull request by RohanMishra315**
#10728

Updated the API documentation of HLRBRep_AlgoPy.xml.

### CODE REVIEW

**Comment by pieterhijma** [comment]
As mentioned in #10728, there are other files that lack documentation as well.

File: src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml [hunk 1]
```diff
@@ -18,89 +18,95 @@
...
         <Documentation>
-            <UserDocu></UserDocu>
+            <UserDocu>Remove the shape S from this framework.</UserDocu>
         </Documentation>
     </Methode>
     <Methode Name="index">
         <Documentation>
-            <UserDocu></UserDocu>
+            <UserDocu>return  the index  of  the  Shape <S>  and
+  return 0 if the Shape <S> is not found.</UserDocu>
```
Comment (chennes) [inline src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml:35]:
```suggestion
            <UserDocu>return  the index  of  the  Shape 'S'  and
  return 0 if the Shape 'S' is not found.</UserDocu>
```

File: src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml [hunk 2]
```diff
@@ -18,89 +18,95 @@
...
         <Documentation>
-            <UserDocu>
+            <UserDocu> set to visible all the edges.
             </UserDocu>
         </Documentation>
     </Methode>
     <Methode Name="hide">
         <Documentation>
-            <UserDocu>
+            <UserDocu>hide the Shape <S> by itself.
```
Comment (chennes) [inline src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml:65]:
```suggestion
            <UserDocu>hide the Shape 'S' by itself.
```

File: src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml [hunk 3]
```diff
@@ -18,89 +18,95 @@
...
             </UserDocu>
         </Documentation>
     </Methode>
     <Methode Name="hideAll">
         <Documentation>
-            <UserDocu>
+            <UserDocu>set to  hide all the  edges of the  Shape <S>.
```
Comment (chennes) [inline src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml:71]:
```suggestion
            <UserDocu>set to  hide all the  edges of the  Shape 'S'.
```

File: src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml [hunk 4]
```diff
@@ -18,89 +18,95 @@
...
         <Documentation>
-            <UserDocu>
+            <UserDocu>select all the DataStructure.
             </UserDocu>
         </Documentation>
     </Methode>
     <Methode Name="selectEdge">
         <Documentation>
-            <UserDocu>
+            <UserDocu>select only the edges of the Shape <S>
```
Comment (chennes) [inline src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml:90]:
```suggestion
            <UserDocu>select only the edges of the Shape 'S'
```

File: src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml [hunk 5]
```diff
@@ -18,89 +18,95 @@
...
             </UserDocu>
         </Documentation>
     </Methode>
     <Methode Name="selectFace">
         <Documentation>
-            <UserDocu>
+            <UserDocu>select only the faces of the Shape <S>.
```
Comment (chennes) [inline src/Mod/Part/App/HLRBRep/HLRBRep_AlgoPy.xml:96]:
```suggestion
            <UserDocu>select only the faces of the Shape 'S'.
```

**Review by chennes — CHANGES_REQUESTED** [review]
As with your other PR, the characters `<` and `>` are not legal in XML, so must be replaced with either their encoded equivalents `&lt;` and `&gt;`, or simply use a different character to indicate your parameter name (I've chosen to use a single quote mark in my suggestions).

**Comment by pieterhijma** [comment]
@luzpaz, @chennes: As promised in https://github.com/FreeCAD/FreeCAD/issues/10728#issuecomment-1795184655, I've worked on this pull request. It is in [branch 'missing-api-doc-py-hlr' on my fork](https://github.com/pieterhijma/FreeCAD/tree/missing-api-doc-py-hlr). Since I didn't initiate this PR, I cannot contribute to this one. I think there are two viable courses of action:
- I create a new pull request and we close this one
- A maintainer takes my commits and forces them onto RohanMishra315:patch-5

My preference is the first option, since I improved on RohanMishra's single commit considerably. By the way, my series of commits builds on RohanMishra commit, so merging the new PR without squashing keeps his work attributed. @RohanMishra315: thanks for starting this work!

For the second option, please note that I rebased my branch on master (including RohanMishra's commit). Please let me know your preference.

**Comment by pieterhijma** [comment]
Commenting on the work that has been done in [branch 'missing-api-doc-py-hlr' on my fork](https://github.com/pieterhijma/FreeCAD/tree/missing-api-doc-py-hlr):

- Since OCCT has a C++ interface, the Python interface needs to be adapted. I carefully checked the types of the Python bindings and matched the documentation accordingly.
- OCCT has lots of missing documentation. I tried to fill in missing pieces as much as possible. As a result, the HLR documentation in FreeCAD's Python interface is now better than OCCT's! (As an example [rg1LineVCompound() in the **poly** algorithm](https://dev.opencascade.org/doc/refman/html/class_h_l_r_b_rep___poly_h_l_r_to_shape.html#a8e64459b306d4d8684766663519f4132) is documented in OCCT but not in the [**non-poly** one](https://dev.opencascade.org/doc/refman/html/class_h_l_r_b_rep___h_l_r_to_shape.html#af39dec75708c85474cdeaf8eac452523). However, they do the same thing, so I took the documentation from the poly algo and reused it in the non-poly algo.)
- Some functions have no documentation. I put a comment in the XML noting that the cause is that OCCT has no documentation.
- I tried to adhere to the standards of types and return values.
- I did the formattting such that it is pleasant in FreeCAD's GUI console.

**Comment by pieterhijma** [comment]
@chennes, ok, thank you, the new PR is #11478.

**Comment by chennes** [comment]
Closing now that #11478 is in.

**Pull request closed without merging**